src/main/java/com/TableTalk/Enterprise/controllers/TableTalkController.java

```diff
@@ -323,31 +324,32 @@ public String login(Model model){
      * @throws Exception
      */
     @RequestMapping("/saveRoom")
-    public String saveRoom(Room room, @RequestParam("imageFile")MultipartFile imageFile, Model model) throws Exception {
-        //todo we shouldn't have to try catch blocks. save everything or save nothing.
+    public ModelAndView saveRoom(Room room, @RequestParam("imageFile")MultipartFile imageFile, Model model) throws Exception {
+        //todo we shouldn't have to try catch blocks. save everything or save nothing. To do that we have to account for
+        // the photo being null
         String returnValue = "createRoom";
-
+        ModelAndView modelAndView = new ModelAndView();
         try {
             roomService.save(room);
         }catch (Exception e) {
             e.printStackTrace();
-            //todo fix this to return an error page
-            return "start";
+            modelAndView.setViewName("error");
+            return  modelAndView;
         }
 
+        Photo photo = new Photo();
         try {
-            Photo photo = new Photo();
             photo.setFilename(imageFile.getOriginalFilename());
-            photo.setPath("/photo/");
             photo.setRoom(room);
             roomService.saveImage(imageFile, photo);
             model.addAttribute("room", room);
-            returnValue = "start";
+            modelAndView.setViewName("success");
         } catch (IOException e){
-            //TODO: change this to logging
-            e.printStackTrace();
-            returnValue = "error";
+            modelAndView.setViewName("error");
```

Comment:
What does the exception say (e)?  Log that.   Or at least, set a breakpoint here and find out.